### DISCUSSION TITLE
audio recording save as .wav format

### DISCUSSION THREAD

**Discussion by ZifengLIU143** [jspsych/jsPsych#3185] [General]
Hi all,

Is it possible to save the audio recording in html-audio-reponse plugin as .wav file directly without further conversion?

Thank you!

Zifeng

**Comment by ChristopheBossens** [comment]
Hi Zifeng,

I think it can always be saved as an audio file but you will also always have to do this by converting the base64 representation because that is how it is stored. How this conversion happens depends on where and how you want to save the file. Do you want to download it when running the jsPsych experiment (you can probably use javascript in this case), or do you want to store it on the server as an audio file (in which case I think you'll need to convert the incoming data). I have always stored the base64 in the data, and used a Python script to convert it to an actual audio file after the experiment.

**Comment by jodeleeuw** [comment]
The format of the base64 audio is based on the browser audio codec, and I don't think .wav is an option for the `MediaRecorder` that gets microphone data. It looks like there might be options to create this functionality (e.g., https://stackoverflow.com/questions/65191193/media-recorder-save-in-wav-format-across-browsers) but we haven't done that yet.
